### ISSUE TITLE
[iOS]Alarm sound plays about 1 second earlier

### ISSUE THREAD

**Issue by r0227n** [gdelataillade/alarm#49]
The scheduling works fine, but on iOS, the alarm sound is played about 1 second earlier than the scheduled time.

※Android works fine.

**Alarm plugin version**
1.0.5

**Describe the bug**
The scheduling works fine, but on iOS, the alarm sound is played about 1 second earlier than the scheduled time.

**To Reproduce**
If I schedule it at any random time, the issue occurs every time.

**Expected behavior**
 Notification sound should occur at the scheduled time.

**Screenshots**

https://github.com/gdelataillade/alarm/assets/40818362/3c097675-91ee-4939-bace-5edb7d996d0e

https://github.com/gdelataillade/alarm/assets/40818362/0e94f4b6-ee99-446e-ae4f-39d6473433a8

**Device info**
- iPhone 14 Pro Max
  - iOS 16.4

**Comment by gdelataillade** [comment]
Hello @r0227n,

I appreciate your patience as we address your concern. The delay you're experiencing on iOS is a result of the plugin checking the alarm time every 500ms. This interval was initially set to optimize application performance and minimize resource usage.

However, I understand that this delay might be affecting your user experience, and I'm open to modifying it. A potential solution could be to reduce the check interval to 200ms. This change should make the alarm trigger more promptly, although it might slightly increase the app's resource usage.

I value your input on this matter. Do you think reducing the check interval to 200ms would be an acceptable solution for your use case?

Looking forward to your response.

**Comment by r0227n** [comment]
Hi @gdelataillade 

I don't possess detailed knowledge about the specific implementation, so my opinion remains abstract.
As there were no perceptible delays on the Android side, I believe it should be fine to use a similar checking interval as android_alarm_manager_plus.

Furthermore, I would like to understand the reasons behind considering 200ms as an appropriate implementation.

Thank you.

**Comment by gdelataillade** [comment]
Hi @r0227n,

For iOS, there is no API similar to `android_alarm_manager_plus` to trigger a callback when alarm rings. When app is foreground, I implemented a simple `Timer.periodic` to check periodically (every 200ms, which explains the delay) if the alarm DateTime is after `DateTime.now`. I chose 200ms to reduce resource usage, but I guess I can lower it to 100ms.

Let me know if you have any question.
